aten/src/ATen/native/native_functions.yaml

```diff
@@ -614,6 +614,9 @@
 - func: diagonal(Tensor(a) self, int offset=0, int dim1=0, int dim2=1) -> Tensor(a)
   variants: function, method
 
+- func: fill_diagonal_(Tensor(a!) self, Scalar fill_value, bool wrap=False) -> Tensor(a!)
+  variants: function, method
```

Comment:
Do we want this to be a function on torch? Originally, we only had in-place operations as methods on torch.Tensor but we've seemed to relax that (perhaps unintentionally)

cc @gchanan @apaszke

Reply:
I like in-place ops being method only.

Reply:
I remove 'function'

Reply:
Yes, it should only be a method.